core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -105,6 +105,17 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val DRIVER_MEMORY_OVERHEAD_FACTOR =
+    ConfigBuilder("spark.driver.memoryOverheadFactor")
+      .doc("The amount of non-heap memory to be allocated per driver in cluster mode, " +
+      "as a fraction of total driver memory. If memory overhead is specified directly, " +
+      "it takes precedence.")
```

Comment:
We can use the current description for `spark.kubernetes.memoryOverheadFactor` in `docs/running-on-kubernetes.md` as template (instead of non-heap memory).

Reply:
Took part of it, let me know how it sounds